### PR TITLE
Checks for publication date in 264 before 260.

### PR DESCRIPTION
**Pull request by shelleydoljack**
Closes https://github.com/traject/traject/issues/224

### CODE REVIEW

File: lib/traject/macros/marc21_semantics.rb [hunk 1]
```diff
@@ -364,13 +364,18 @@ def self.publication_date(record, estimate_tolerance = 15, min_year = 500, max_y
...
         v260c = MarcExtractor.cached("260c", :separator => nil).extract(record).first
         # just try to take the first four digits out of there, we're not going to try
         # anything crazy.
-        if m = /(\d{4})/.match(v260c)
+        if m = /(\d{4})/.match(v264c)
           found_date = m[1].to_i
+        else
+          if m = /(\d{4})/.match(v260c)
+            found_date = m[1].to_i
+          end
```
Comment (jrochkind) [inline lib/traject/macros/marc21_semantics.rb]: Minor stylistic thing, but can you change this to `elsif` without the separate inner block? I think that would do the same thing? This is already convoluted logic, so let's try to keep it as simple as we can!

**Comment by jrochkind** [comment]
Ah, and I was wrong, it's not looking at 264 regardless of indicators -- it looks like it's only willing to use a 264 with second indicator 1 -- `publication`. It ignores 264s with second indicator meaning production, distribution, manufacture, or copyright.

This is what you intended @shelleydoljack ? Maybe the comment you added about 264, expand it to say "264 with second indicator 1 for 'publication'"?

I think this is necessarily going to be a 'best guess common logic', which can't work universally, some people will need to write custom logic. But you think ignoring 264 unless second indicator 1 publication makes sense?   For 260, the best-guess data extraction logic is willing to take _any_ 260 -- so if it doesn't find a 264 second indicator 1, but does find a 260, it'll just use it. Should it be willing to take any 264 too, or is the way it is now in PR best?

https://www.oclc.org/bibformats/en/2xx/260.html

**Comment by shelleydoljack** [comment]
@jrochkind If it should be "best guess common logic", then it's probably best to look for any 264 field (not just those with 2nd indicator 1). I will make this change.

**Comment by jrochkind** [comment]
Cool! Does that still work for you @shelleydoljack , or will that make you not want to use the logic? I'm honestly not sure what is best, just think it's worth asking the question and considering it a bit!

**Comment by jrochkind** [comment]
thanks @shelleydoljack! 

Ordinarily a traject release with this change might not happen for a while, until some other changes pile up too. Let me know if that's inconvenient for you and you would like a sooner release with just this change!

**Pull request merged**